ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -660,17 +660,38 @@ let workspace_symbol server (state : State.t) (params : WorkspaceSymbolParams.t)
     let+ symbols_results =
       let task =
         Lev_fiber.Thread.task thread ~f:(fun () ->
+            let+ () = Fiber.return () in
             Workspace_symbol.run params workspaces)
       in
-      let+ res =
+      let* res, cancel =
         match task with
-        | Ok task -> Lev_fiber.Thread.await task
+        | Ok task ->
+          let* cancel = Server.cancel_token () in
+          let maybe_cancel =
+            match cancel with
+            | None ->
+              fun f ->
+                let+ res = f () in
+                (res, Fiber.Cancel.Not_cancelled)
+            | Some token ->
+              let on_cancel () = Lev_fiber.Thread.cancel task in
```

Comment:
That looks good, but I think we can do even better. Why don't we check if the cancellation token has been fired before reading every cmt file? If it has been fired, we can terminate the work early.